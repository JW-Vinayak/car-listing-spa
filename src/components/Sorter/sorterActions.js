export const SET_SORT_ORDER = "SET_SORT_ORDER";

export const setSortOrder = sortOrder => ({
  type: SET_SORT_ORDER,
  payload: { sortOrder }
});
