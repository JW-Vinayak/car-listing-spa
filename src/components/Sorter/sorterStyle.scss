.sorter {
  .sort-dropdown {
    width: 200px;
  }
  .sort-by-text {
    font-size:0.9em;
    margin-bottom: $margin-8;
  }
  float:right;
}
