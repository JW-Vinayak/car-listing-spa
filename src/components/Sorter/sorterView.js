import React, { useState } from "react";
import PropTypes from "prop-types";

import {
  NO_MILEAGE_SORTING,
  ASCENDING_MILEAGE_SORTING,
  DESCENDING_MILEAGE_SORTING
} from "../../common/constants";

const Sorter = props => {
  let [sortOrder, setSortOrder] = useState(props.sortOrder);
  let { setSortOrderInStore } = props;

  let handleChange = event => {
    console.log("sort value", event.target.value);
    setSortOrder(event.target.value);
    setSortOrderInStore(event.target.value);
  };

  return (
    <div className="sorter">
      <p className="sort-by-text">Sort by</p>
      <select
        className="filter-dropdown sort-dropdown"
        value={sortOrder}
        onChange={e => handleChange(e)}
      >
        <option value={NO_MILEAGE_SORTING}>None</option>
        <option value={ASCENDING_MILEAGE_SORTING}>Mileage Ascending</option>
        <option value={DESCENDING_MILEAGE_SORTING}>Mileage Descending</option>
      </select>
    </div>
  );
};

Sorter.propTypes = {
  setSortOrderInStore: PropTypes.func.isRequired,
  sortOrder: PropTypes.string.isRequired
};

export default Sorter;
