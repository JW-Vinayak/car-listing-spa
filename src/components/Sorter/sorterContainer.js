import { connect } from 'react-redux';
import { setSortOrder } from './sorterActions';
import SorterView from './sorterView'

const mapStateToProps = state => ({
  sortOrder: state.sorter.sortOrder
})

const mapDispatchToProps = dispatch => ({
  setSortOrderInStore: (sortOrder) => dispatch(setSortOrder(sortOrder))
})

export default connect(mapStateToProps, mapDispatchToProps)(SorterView)
