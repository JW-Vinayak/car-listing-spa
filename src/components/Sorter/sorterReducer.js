import { NO_MILEAGE_SORTING } from '../../common/constants';
import { SET_SORT_ORDER } from './sorterActions';

const initialState = {
  mileage: NO_MILEAGE_SORTING
}

export default function sortReducer(state = initialState, action) {
  switch (action.type) {
    case SET_SORT_ORDER:
      return {
        ...state,
        mileage: action.payload.sortOrder
      }
    default:
      return {
        ...state
      }

  }
}
