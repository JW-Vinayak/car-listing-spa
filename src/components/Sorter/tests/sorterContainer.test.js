import React from "react";
import thunk from "redux-thunk";
import Sorter from "../sorterContainer";
import configureMockStore from "redux-mock-store";
import renderer from "react-test-renderer";

const middlewares = [thunk];

const mockStore = configureMockStore(middlewares);

let store = mockStore({
  sorter: {
    mileage: "NO_MILEAGE_SORTING"
  }
});

describe("Sorter component", () => {
  it("shallow renders without crashing", () => {
    const tree = renderer.create(<Sorter store={store} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
