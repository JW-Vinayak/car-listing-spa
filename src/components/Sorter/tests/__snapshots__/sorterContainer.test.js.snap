// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorter component shallow renders without crashing 1`] = `
<div
  className="sorter"
>
  <p
    className="sort-by-text"
  >
    Sort by
  </p>
  <select
    className="filter-dropdown sort-dropdown"
    onChange={[Function]}
    value="NO_MILEAGE_SORTING"
  >
    <option
      value="NO_MILEAGE_SORTING"
    >
      None
    </option>
    <option
      value="ASCENDING_MILEAGE_SORTING"
    >
      Mileage Ascending
    </option>
    <option
      value="DESCENDING_MILEAGE_SORTING"
    >
      Mileage Descending
    </option>
  </select>
</div>
`;
